### PR TITLE
docs: Add `@method` for `when()` and `whenNot()` in models.

### PR DESCRIPTION
**Pull request by sammyskills**
**Description**
The `when()` and `whenNot()` conditional statements works directly with models, i.e., one does not need to call the `builder()` method before using it. So, they should be type-hinted in IDEs or code editors, just like others. 
See https://forum.codeigniter.com/showthread.php?tid=88190&pid=411611#pid411611

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by ddevsr — APPROVED** [review]

File: tests/system/Models/WhenWhenNotModelTest.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+use Tests\Support\Models\SecondaryModel;
+
+/**
+ * @group DatabaseLive
+ *
+ * @internal
+ */
+final class WhenWhenNotModelTest extends LiveModelTestCase
+{
+    public function testWhenWithSingleCallback(): void
```
Comment (kenjis) [inline tests/system/Models/WhenWhenNotModelTest.php:23]: This method name is difficult to understand what is tested.
E.g., `testWhenWithTrueCondition()`?

File: tests/system/Models/WhenWhenNotModelTest.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        $result = $this->model->when($filter, static function ($query, $filter) {
+            $query->where('value', $filter);
+        })->find();
+
+        $this->assertCount(2, $result);
+        $this->assertSame('foo', $result[0]->key);
+        $this->assertSame('bar', $result[1]->key);
+    }
+
+    public function testWhenWithBothCallbacks(): void
```
Comment (kenjis) [inline tests/system/Models/WhenWhenNotModelTest.php:52]:
```suggestion
    public function testWhenWithFalseConditionAndDefaultCallback(): void
```

File: tests/system/Models/WhenWhenNotModelTest.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        }, static function ($query) {
+            $query->where('value', 'foobar');
+        })->find();
+
+        $this->assertCount(2, $result);
+        $this->assertSame('foo', $result[0]->key);
+        $this->assertSame('bar', $result[1]->key);
+    }
+
+    public function testWhenNotWithSingleCallback(): void
```
Comment (kenjis) [inline tests/system/Models/WhenWhenNotModelTest.php:83]:
```suggestion
    public function testWhenNotWithFalseCondition(): void
```

File: tests/system/Models/WhenWhenNotModelTest.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        $result = $this->model->whenNot($filter, static function ($query, $filter) {
+            $query->where('value !=', 'foobar');
+        })->find();
+
+        $this->assertCount(1, $result);
+        $this->assertSame('baz', $result[0]->key);
+        $this->assertSame('foobaz', $result[0]->value);
+    }
+
+    public function testWhenNotWithBothCallbacks(): void
```
Comment (kenjis) [inline tests/system/Models/WhenWhenNotModelTest.php:112]:
```suggestion
    public function testWhenNotWithTrueConditionAndDefaultCallback(): void
```

File: tests/system/Models/WhenWhenNotModelTest.php (hunk 1 above)
Comment (kenjis) [inline tests/system/Models/WhenWhenNotModelTest.php:23]: This method name is difficult to understand what is tested.
```suggestion
    public function testWhenWithTrueCondition(): void
```

**Review by MGatner — APPROVED** [review]
Looks good! Rector failure is a mistake - rerunning.

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@sammyskills Thank you.

We recommend `git rebase` instead of `git merge` to update PR branches.
I hope you will be able to use it.
See https://github.com/codeigniter4/shield/pull/789#issuecomment-1684371442

**Pull request merged**